x-pack/plugins/observability_solution/infra/server/lib/alerting/inventory_metric_threshold/inventory_metric_threshold_executor.ts

```diff
@@ -59,26 +58,10 @@ type InventoryMetricThresholdAllowedActionGroups = ActionGroupIdsOf<
 export const FIRED_ACTIONS_ID = 'metrics.inventory_threshold.fired';
 export const WARNING_ACTIONS_ID = 'metrics.inventory_threshold.warning';
 
-type InventoryThrehsoldActionGroup = typeof FIRED_ACTIONS_ID | typeof WARNING_ACTIONS_ID;
-
 export type InventoryMetricThresholdRuleTypeState = RuleTypeState; // no specific state used
 export type InventoryMetricThresholdAlertState = AlertState; // no specific state used
 export type InventoryMetricThresholdAlertContext = AlertContext; // no specific instance context used
 
-type InventoryMetricThresholdAlert = Alert<
-  InventoryMetricThresholdAlertState,
-  InventoryMetricThresholdAlertContext,
-  InventoryMetricThresholdAllowedActionGroups
->;
-type InventoryMetricThresholdAlertFactory = (
-  id: string,
-  reason: string,
-  actionGroup: InventoryThrehsoldActionGroup,
-  additionalContext?: AdditionalContext | null,
-  evaluationValues?: Array<number | null>,
-  thresholds?: Array<number | null>
-) => InventoryMetricThresholdAlert;
-
 export const createInventoryMetricThresholdExecutor = (libs: InfraBackendLibs) =>
   libs.metricsRules.createLifecycleRuleExecutor<
```

Comment:
we need to remove this usage of `createLifecycleRuleExecutor` so that this no longer goes through the rule registry

Reply:
I totally missed that! Thanks for pointing out.
Fixed with  https://github.com/elastic/kibana/pull/178506/commits/5c5315fcc75980893c85ee8050794d5707795eac